<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QtMainWindow</class>
 <widget class="QMainWindow" name="QtMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1631</width>
    <height>627</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QtMainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="btn_logout">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>10</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>logout</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="table_musiclist">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>60</y>
      <width>441</width>
      <height>491</height>
     </rect>
    </property>
    <property name="selectionBehavior">
     <enum>QAbstractItemView::SelectRows</enum>
    </property>
    <column>
     <property name="text">
      <string>곡명</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>아티스트</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>장르</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>작곡가</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>곡 ID</string>
     </property>
    </column>
   </widget>
   <widget class="QLabel" name="label_loginInfo">
    <property name="geometry">
     <rect>
      <x>850</x>
      <y>10</y>
      <width>261</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Logged in as:</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QSlider" name="horizontalSlider">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>180</y>
      <width>301</width>
      <height>22</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="ClickableLabel" name="btn_player_prev">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>220</y>
      <width>61</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap>images/baseline_skip_previous_black_48dp.png</pixmap>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="ClickableLabel" name="btn_player_play">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>220</y>
      <width>61</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap>images/baseline_play_arrow_black_48dp.png</pixmap>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="ClickableLabel" name="btn_player_next">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>220</y>
      <width>61</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap>images/baseline_skip_next_black_48dp.png</pixmap>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_playerAlbum">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>60</y>
      <width>91</width>
      <height>91</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap>images/3825045661552302096-256.png</pixmap>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QToolButton" name="btn_musicList_add">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>560</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Add music to list</string>
    </property>
    <property name="popupMode">
     <enum>QToolButton::MenuButtonPopup</enum>
    </property>
    <property name="arrowType">
     <enum>Qt::NoArrow</enum>
    </property>
   </widget>
   <widget class="QTableWidget" name="table_playlist_current">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>320</y>
      <width>341</width>
      <height>231</height>
     </rect>
    </property>
    <property name="editTriggers">
     <set>QAbstractItemView::NoEditTriggers</set>
    </property>
    <property name="selectionBehavior">
     <enum>QAbstractItemView::SelectRows</enum>
    </property>
    <column>
     <property name="text">
      <string>곡명</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>아티스트</string>
     </property>
    </column>
   </widget>
   <widget class="QLabel" name="label_playerTitle">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>70</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>대충 노래 제목</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_playerArtist">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>100</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>대충 가수명</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="text_lyrics">
    <property name="geometry">
     <rect>
      <x>1270</x>
      <y>60</y>
      <width>331</width>
      <height>491</height>
     </rect>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Gulim'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_showLyrics">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>560</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Show lyrics &gt;&gt;</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_searchLyrics">
    <property name="geometry">
     <rect>
      <x>1480</x>
      <y>560</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Search for lyrics</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_musicList_remove">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>560</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="table_playlist">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>60</y>
      <width>341</width>
      <height>491</height>
     </rect>
    </property>
    <property name="editTriggers">
     <set>QAbstractItemView::NoEditTriggers</set>
    </property>
    <property name="selectionBehavior">
     <enum>QAbstractItemView::SelectRows</enum>
    </property>
    <column>
     <property name="text">
      <string>곡명</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>아티스트</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>재생 횟수</string>
     </property>
    </column>
   </widget>
   <widget class="QPushButton" name="btn_playlist_remove">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>560</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
   <widget class="QToolButton" name="btn_playlist_play">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>560</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>▶ Play</string>
    </property>
    <property name="popupMode">
     <enum>QToolButton::MenuButtonPopup</enum>
    </property>
    <property name="arrowType">
     <enum>Qt::NoArrow</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_saveLyrics">
    <property name="geometry">
     <rect>
      <x>1350</x>
      <y>560</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Save lyrics</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>30</y>
      <width>441</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>♬ Music list</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>30</y>
      <width>341</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>♬ Playlist</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>290</y>
      <width>341</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>▶ Now playing</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>1270</x>
      <y>30</y>
      <width>331</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string> Lyrics</string>
    </property>
   </widget>
   <widget class="ClickableLabel" name="btn_searchMusic">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>560</y>
      <width>31</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap>images/baseline_search_black_24dp.png</pixmap>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="ClickableLabel" name="btn_lockPlaylist">
    <property name="geometry">
     <rect>
      <x>850</x>
      <y>35</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="pixmap">
     <pixmap>images/baseline_lock_black_24dp.png</pixmap>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_playlist_explore">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>560</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Explore</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_musicListToPlaylist">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>260</y>
      <width>41</width>
      <height>91</height>
     </rect>
    </property>
    <property name="text">
     <string>&gt;&gt;</string>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ClickableLabel</class>
   <extends>QLabel</extends>
   <header>ClickableLabel.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>btn_logout</sender>
   <signal>clicked()</signal>
   <receiver>QtMainWindow</receiver>
   <slot>LogoutClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1140</x>
     <y>32</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>161</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_musicList_add</sender>
   <signal>clicked()</signal>
   <receiver>QtMainWindow</receiver>
   <slot>AddMusicPopup()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>582</y>
    </hint>
    <hint type="destinationlabel">
     <x>214</x>
     <y>602</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_showLyrics</sender>
   <signal>clicked()</signal>
   <receiver>QtMainWindow</receiver>
   <slot>ShowLyricsClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1142</x>
     <y>574</y>
    </hint>
    <hint type="destinationlabel">
     <x>870</x>
     <y>614</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_playlist_play</sender>
   <signal>clicked()</signal>
   <receiver>QtMainWindow</receiver>
   <slot>PlayFromStartClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>683</x>
     <y>574</y>
    </hint>
    <hint type="destinationlabel">
     <x>562</x>
     <y>605</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>table_playlist</sender>
   <signal>cellDoubleClicked(int,int)</signal>
   <receiver>QtMainWindow</receiver>
   <slot>PlaylistCellDoubleClicked(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>687</x>
     <y>221</y>
    </hint>
    <hint type="destinationlabel">
     <x>727</x>
     <y>47</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>table_playlist_current</sender>
   <signal>cellDoubleClicked(int,int)</signal>
   <receiver>QtMainWindow</receiver>
   <slot>CurPlaylistCellDoubleClicked(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1016</x>
     <y>435</y>
    </hint>
    <hint type="destinationlabel">
     <x>972</x>
     <y>315</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_searchLyrics</sender>
   <signal>clicked()</signal>
   <receiver>QtMainWindow</receiver>
   <slot>SearchForLyricsClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1517</x>
     <y>575</y>
    </hint>
    <hint type="destinationlabel">
     <x>1389</x>
     <y>606</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_saveLyrics</sender>
   <signal>clicked()</signal>
   <receiver>QtMainWindow</receiver>
   <slot>SaveLyricsClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1370</x>
     <y>570</y>
    </hint>
    <hint type="destinationlabel">
     <x>1269</x>
     <y>626</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_playlist_remove</sender>
   <signal>clicked()</signal>
   <receiver>QtMainWindow</receiver>
   <slot>RemoveFromPlaylistClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>855</x>
     <y>575</y>
    </hint>
    <hint type="destinationlabel">
     <x>783</x>
     <y>626</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_musicList_remove</sender>
   <signal>clicked()</signal>
   <receiver>QtMainWindow</receiver>
   <slot>RemoveFromMusicListClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>467</x>
     <y>577</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>607</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_playlist_explore</sender>
   <signal>clicked()</signal>
   <receiver>QtMainWindow</receiver>
   <slot>ExplorePlaylistClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>670</x>
     <y>583</y>
    </hint>
    <hint type="destinationlabel">
     <x>935</x>
     <y>598</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>table_musiclist</sender>
   <signal>cellChanged(int,int)</signal>
   <receiver>QtMainWindow</receiver>
   <slot>MusicListCellChanged(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>160</x>
     <y>488</y>
    </hint>
    <hint type="destinationlabel">
     <x>145</x>
     <y>579</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_musicListToPlaylist</sender>
   <signal>clicked()</signal>
   <receiver>QtMainWindow</receiver>
   <slot>MusicListToPlaylistClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>500</x>
     <y>287</y>
    </hint>
    <hint type="destinationlabel">
     <x>500</x>
     <y>378</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>Logout()</slot>
  <slot>FinishProgram()</slot>
  <slot>AddMusicPopup()</slot>
  <slot>MusicCellDoubleClicked(int,int)</slot>
  <slot>LogoutClicked()</slot>
  <slot>ShowLyricsClicked()</slot>
  <slot>PlayFromStartClicked()</slot>
  <slot>PlaylistCellDoubleClicked(int,int)</slot>
  <slot>CurPlaylistCellDoubleClicked(int,int)</slot>
  <slot>SearchForLyricsClicked()</slot>
  <slot>SaveLyricsClicked()</slot>
  <slot>RemoveFromPlaylistClicked()</slot>
  <slot>RemoveFromMusicListClicked()</slot>
  <slot>ExplorePlaylistClicked()</slot>
  <slot>MusicListCellChanged(int,int)</slot>
  <slot>MusicListToPlaylistClicked()</slot>
 </slots>
</ui>
